### PR TITLE
Update to v2.2.0

### PR DESCRIPTION
**Pull request by traversaro**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by traversaro** [comment]
@conda-forge-admin, please rerender

**Comment by traversaro** [comment]
The build is failing with error:
~~~
[ 45%] Building Fortran object CMakeFiles/scalapack.dir/SRC/pstrsen.f.o
/home/conda/feedstock_root/build_artifacts/scalapack_1646690481794/work/SRC/pstrsen.f:608:48:

  529 |      $           CALL IGAMX2D( ICTXT, 'All', TOP, 1, 1, MMAX, 1, -1,
      |                                                      2
......
 3253 |             CALL IGAMX2D( ICTXT, 'All', TOP, 1, 1, IERR, 1, -1,
      |                                                   1
Error: Rank mismatch between actual argument at (1) and actual argument at (2) (rank-1 and scalar)
/home/conda/feedstock_root/build_artifacts/scalapack_1646690481794/work/SRC/pstrord.f:3264:54:

  490 |      $         CALL IGAMX2D( ICTXT, 'All', TOP, 1, 1, MMAX, 1, -1,
      |                                                      2
......
 3264 |                CALL IGAMX2D( ICTXT, 'All', TOP, 1, 1, INFO, 1, -1,
      |                                                      1
Error: Rank mismatch between actual argument at (1) and actual argument at (2) (rank-1 and scalar)
make[2]: *** [CMakeFiles/scalapack.dir/build.make:12812: CMakeFiles/scalapack.dir/SRC/pstrord.f.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:1234: CMakeFiles/scalapack.dir/all] Error 2
make: *** [Makefile:166: all] Error 2
Traceback (most recent call last):
~~~

This is a gcc 10 compatibility problem that should have been fixed in https://github.com/Reference-ScaLAPACK/scalapack/pull/23, but this apparently is not the case. Indeed, the same compilation error appeared in homebrew as well, see https://github.com/Homebrew/homebrew-core/pull/94420/files#diff-e45b9bfcc61ceafda0d6200c3c06dec48c0da5a6259d8346659d3fe822b0ae0bL35 .

Issue reported upstream in https://github.com/Reference-ScaLAPACK/scalapack/issues/55 .

**Comment by traversaro** [comment]
@conda-forge-admin, please rerender

**Comment by traversaro** [comment]
Now compilation fails with:
~~~
2022-03-07T22:38:10.6429303Z /home/conda/feedstock_root/build_artifacts/scalapack_1646692393731/_build_env/bin/../lib/gcc/x86_64-conda-linux-gnu/10.3.0/../../../../x86_64-conda-linux-gnu/bin/ld: CMakeFiles/xshseqr.dir/pshseqrdriver.f.o: in function `MAIN__':
2022-03-07T22:38:10.6434356Z pshseqrdriver.f:(.text.MAIN__+0x104): undefined reference to `mpi_wtime_'
2022-03-07T22:38:10.6486751Z /home/conda/feedstock_root/build_artifacts/scalapack_1646692393731/_build_env/bin/../lib/gcc/x86_64-conda-linux-gnu/10.3.0/../../../../x86_64-conda-linux-gnu/bin/ld: pshseqrdriver.f:(.text.MAIN__+0x1ab): undefined reference to `mpi_wtime_'
2022-03-07T22:38:10.6489441Z /home/conda/feedstock_root/build_artifacts/scalapack_1646692393731/_build_env/bin/../lib/gcc/x86_64-conda-linux-gnu/10.3.0/../../../../x86_64-conda-linux-gnu/bin/ld: pshseqrdriver.f:(.text.MAIN__+0x4ae): undefined reference to `mpi_wtime_'
2022-03-07T22:38:10.6491257Z /home/conda/feedstock_root/build_artifacts/scalapack_1646692393731/_build_env/bin/../lib/gcc/x86_64-conda-linux-gnu/10.3.0/../../../../x86_64-conda-linux-gnu/bin/ld: pshseqrdriver.f:(.text.MAIN__+0x519): undefined reference to `mpi_wtime_'
2022-03-07T22:38:10.6492995Z /home/conda/feedstock_root/build_artifacts/scalapack_1646692393731/_build_env/bin/../lib/gcc/x86_64-conda-linux-gnu/10.3.0/../../../../x86_64-conda-linux-gnu/bin/ld: pshseqrdriver.f:(.text.MAIN__+0x556): undefined reference to `mpi_wtime_'
2022-03-07T22:38:10.6494887Z /home/conda/feedstock_root/build_artifacts/scalapack_1646692393731/_build_env/bin/../lib/gcc/x86_64-conda-linux-gnu/10.3.0/../../../../x86_64-conda-linux-gnu/bin/ld: CMakeFiles/xshseqr.dir/pshseqrdriver.f.o:pshseqrdriver.f:(.text.MAIN__+0x55e): more undefined references to `mpi_wtime_' follow
2022-03-07T22:38:10.6495980Z collect2: error: ld returned 1 exit status
~~~
On Linux and an macOS:
~~~
$BUILD_PREFIX/bin/x86_64-apple-darwin13.4.0-gfortran   -march=core2 -mtune=haswell -ftree-vectorize -fPIC -fstack-protector -O2 -pipe -isystem $PREFIX/include -fdebug-prefix-map=$SRC_DIR=/usr/local/src/conda/scalapack-2.2.0 -fdebug-prefix-map=$PREFIX=/usr/local/src/conda-prefix -fallow-argument-mismatch  -isysroot /Applications/Xcode_12.4.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.9.sdk -mmacosx-version-min=10.9 -c $SRC_DIR/build/CMakeFiles/CMakeTmp/testFortranCompiler.f -o CMakeFiles/cmTC_13de7.dir/testFortranCompiler.f.o
x86_64-apple-darwin13.4.0-gfortran: error: unrecognized command line option ‘-fallow-argument-mismatch’; did you mean ‘-Wno-argument-mismatch’?
~~~

As the CMake build system does not correctly link mpi library, better to just use the mpi wrapper instead of the normal compilers.

**Comment by traversaro** [comment]
> This is a gcc 10 compatibility problem that should have been fixed in [Reference-ScaLAPACK/scalapack#23](https://github.com/Reference-ScaLAPACK/scalapack/pull/23), but this apparently is not the case. Indeed, the same compilation error appeared in homebrew as well, see https://github.com/Homebrew/homebrew-core/pull/94420/files#diff-e45b9bfcc61ceafda0d6200c3c06dec48c0da5a6259d8346659d3fe822b0ae0bL35 .

The workaround for this is the `-fallow-argument-mismatch` gfortran option added in https://github.com/conda-forge/scalapack-feedstock/pull/30/commits/a0594a58521bc6223ee9ecf2fbdd2bc800fd7d30, but we should add some logic as macos still uses gfortran 9.

**Comment by traversaro** [comment]
CI is happy, @conda-forge/scalapack the PR is ready for review. If you like, I can add myself to the mantainers team.

**Comment by Tobias-Fischer** [comment]
Awesome work, thanks @traversaro!

**Review by jschueller — APPROVED** [review]
+1

**Comment by dalcinl** [comment]
@traversaro If you are interested, please add yourself as a recipe maintainer, then I'll merge.

**Comment by traversaro** [comment]
> @traversaro If you are interested, please add yourself as a recipe maintainer, then I'll merge.

Done, thanks: https://github.com/conda-forge/scalapack-feedstock/pull/30/commits/b837aa29aac0b1d854f6d2632ec2810e754fd755 .

File: recipe/build.sh [hunk 1]
```diff
@@ -8,8 +8,20 @@ if [[ "$target_platform" == "osx-64" ]]; then
   fi
 fi
 
+if [ "$(expr substr $(uname -s) 1 5)" == "Linux" ]
```
Comment (dalcinl) [inline recipe/build.sh]: Just curious: why not just `if [ "$(uname -s)" == "Linux" ]` ?

Comment (traversaro) [reply recipe/build.sh]: Good point, I probably just copy pasted a bit to blindly from some resource online. `uname -s` is probably enough.

Comment (traversaro) [reply recipe/build.sh]: See https://stackoverflow.com/questions/3466166/how-to-check-if-running-in-cygwin-mac-or-linux .

Comment (dalcinl) [reply recipe/build.sh]: Actually, should't it be better if we do: `if [[ "$target_platform" == linux-* ]]` ?

Comment (traversaro) [reply recipe/build.sh]: Sure, that is definetely a better idea, let's use that.

Comment (traversaro) [reply recipe/build.sh]: Done in https://github.com/conda-forge/scalapack-feedstock/pull/30/commits/ef79d5ca658c338e59416dba8ed314aeaf003fd9 .

**Pull request merged**